sdk/schemaregistry/schema-registry-avro/src/schemaRegistryAvroSerializer.ts

```diff
@@ -195,11 +199,18 @@ export class SchemaRegistryAvroSerializer {
       content: schema
     };
 
-    const schemaIdResponse = this.autoRegisterSchemas
-      ? await this.registry.registerSchema(description)
-      : await this.registry.getSchemaId(description);
+    let id: string;
+    if (this.autoRegisterSchemas) {
+      id = (await this.registry.registerSchema(description)).id;
+    } else {
+      const response = await this.registry.getSchemaId(description);
+      if (!response) {
+        throw new Error("Schema not found in registry.");
```

Comment:
is there any meaningful way we can refer to the schema to provide better logging?

Reply:
Yes, there is a name I should have used. Will fix.

Reply:
It could be confusing though because the name could match but the content differs and this throws. Will think about how to word this.

Reply:
I'm thinking something like "Schema with name <name> not found in registry or not found with matching content." A little awkward, so any thoughts on a better wording are most welcome.

Reply:
I think content would be too verbose, though if you *really* wanted to you could make a custom error type to attach it as a property. Not really worth it if it's not being programmatically handled though.